OpenRA.Mods.Cnc/Activities/LayMines.cs

```diff
@@ -118,17 +122,17 @@ public void CleanMineField(Actor self)
 		public override IEnumerable<TargetLineNode> TargetLineNodes(Actor self)
 		{
 			if (returnToBase)
-				yield return new TargetLineNode(Target.FromActor(rearmTarget), Color.Green);
+				yield return new TargetLineNode(Target.FromActor(rearmTarget), mobile == null ? aircraft.Info.TargetLineColor : mobile.Info.TargetLineColor);
```

Comment:
Crashing here when neither `mobile` nor `aircraft` are used as `IMove` by the actor looks rather odd to me. What do you think about having `IMoveInfo` define a `GetTargetLineColor` method which is overwritten/implemented by `MobileInfo` and `AircraftInfo` and called here instead of "hardcoding" specific traits?

Reply:
Good idea! fixed.